examples/chip-tool/commands/pairing/PairingCommand.cpp

```diff
@@ -404,8 +416,26 @@ void PairingCommand::OnICDRegistrationComplete(NodeId nodeId, uint32_t icdCounte
     chip::Encoding::BytesToHex(mICDSymmetricKey.Value().data(), mICDSymmetricKey.Value().size(), icdSymmetricKeyHex,
                                sizeof(icdSymmetricKeyHex), chip::Encoding::HexFlags::kNullTerminate);
 
-    // TODO: Persist symmetric key.
+    chip::app::ICDClientInfo clientInfo;
+    clientInfo.peer_node         = chip::ScopedNodeId(nodeId, CurrentCommissioner().GetFabricIndex());
+    clientInfo.monitored_subject = mICDMonitoredSubject.Value();
+
+    CHIP_ERROR err = GetICDClientStorage().SetKey(clientInfo, mICDSymmetricKey.Value());
+    if (err == CHIP_NO_ERROR)
+    {
+        err = GetICDClientStorage().StoreEntry(clientInfo);
+    }
+
+    if (err != CHIP_NO_ERROR)
+    {
+        // StoreEntry is unlikely to fail since it is a local operation for CHIP Tool.
+        ChipLogError(chipTool, "Failed to persist symmetric key for " ChipLogFormatX64 ": %s", ChipLogValueX64(nodeId),
+                     ErrorStr(err));
```

Comment:
```suggestion
                     err.AsString());
```

Reply:
But why?

Should we also replace existing use of `ErrorStr` by `AsString`?